tests/system/action/user/test_account_import.py

```diff
@@ -0,0 +1,760 @@
+                                    }
+                                },
+                            },
+                            {
+                                "state": ImportState.DONE,
+                                "messages": [],
+                                "data": {
+                                    "username": {
+                                        "value": "durban",
+                                        "info": ImportState.DONE,
+                                    }
+                                },
+                            },
+                        ],
+                    },
+                }
+            }
+        )
+        response = self.request("account.import", {"id": 7, "import": True})
+        self.assert_status_code(response, 200)
+        result = response.json["results"][0][0]
+        assert result["state"] == ImportState.ERROR
+        for i in range(2):
+            entry = result["rows"][i]
+            assert entry["state"] == ImportState.ERROR
+            assert entry["messages"] == [
+                "Error: username 'durban' is duplicated in import."
+            ]
+
+    def test_import_error_state_done_not_matching_ids(self) -> None:
+        self.set_models(
+            {
+                "user/8": {"username": "user8"},
+                **self.get_import_preview_data(
+                    6,
+                    ImportState.DONE,
+                    {
+                        "id": 5,
+                        "first_name": "Testy",
+                        "username": {
+                            "value": "user8",
+                            "info": ImportState.DONE,
+                            "id": 5,
+                        },
+                    },
+                ),
+            }
+        )
+        response = self.request("account.import", {"id": 6, "import": True})
+        self.assert_status_code(response, 200)
+        entry = response.json["results"][0][0]["rows"][0]
+        assert entry["state"] == ImportState.ERROR
+        assert entry["messages"] == [
+            "Error: username 'user8' found in different id (8 instead of 5)"
+        ]
+
+    def test_import_error_state_import_preview4(self) -> None:
+        response = self.request("account.import", {"id": 4, "import": True})
+        self.assert_status_code(response, 400)
+        assert response.json["message"] == "Error in import. Data will not be imported."
+        self.assert_model_exists("import_preview/4")
+
+    def test_import_no_permission(self) -> None:
+        self.base_permission_test({}, "account.import", {"id": 2, "import": True})
+
+    def test_import_permission(self) -> None:
+        self.base_permission_test(
+            {},
+            "account.import",
+            {"id": 2, "import": True},
+            OrganizationManagementLevel.CAN_MANAGE_USERS,
+        )
+
+
+class AccountJsonImportWithIncludedJsonUpload(AccountJsonUploadForUseInImport):
```

Comment:
I would prefer to keep the json-upload tests all in the test_json_upload file, because they were developed as real tests and act as such. They are not meant or written as setup.
I want to use them as input for the import-tests, becaue

1. the correct structure is guaranteed 
2. factual tests of upload could be easily used for the same factual test in import
3. If needed it is easy in split screen mode to see the json upload data and the import data side by side  

 The removal of the test prefix is just done to prevent a duplicate execution of the same test.

Reply:
I'm not against the idea of using the tests as setup for the import tests. They don't need to be written as setup for them to be moved into a separate file, e.g. `test_json_upload_and_import`. This would make the whole structure easier to understand, and also why the prefix is not there. You don't have to explain it to me why the prefix is missing, but to someone who looks at these tests the first time. Having a clear separation between upload tests which are only used as upload tests and upload tests which act as setup as well should be beneficial.